File: Examples/CSharp/WorkingWithResources/GetResourcePercentWorkComplete.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithResources
{
    using System;

    internal class GetResourcePercentWorkComplete
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:GetResourcePercentWorkComplete
            //ExFor: Rsc.PercentWorkComplete
            //ExSummary: Shows how to read resource percent work complete.
            var project = new Project(dataDir + "ResourcePercentWorkComplete.mpp");

            // Display work percentage completion for all resources
            foreach (var res in project.Resources)
            {
                if (res.Get(Rsc.Name) != null)
                {
                    Console.WriteLine(res.Get(Rsc.PercentWorkComplete));
                }
            }
            //ExEnd:GetResourcePercentWorkComplete
        }
    }
}

File: Examples/VisualBasic/WorkingWithResources/GetResourcePercentWorkComplete.vb

Namespace WorkingWithResources
    Public Class GetResourcePercentWorkComplete
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:GetResourcePercentWorkComplete
            ' Create project instance
            Dim project1 As New Project(dataDir + "ResourcePercentWorkComplete.mpp")

            ' Display work percentage completion for all resources
            For Each res As Resource In project1.Resources
                If res.[Get](Rsc.Name) IsNot Nothing Then
                    Console.WriteLine(res.[Get](Rsc.PercentWorkComplete))
                End If
            Next
            ' ExEnd:GetResourcePercentWorkComplete
        End Sub
    End Class
End Namespace

